src/mixins/onThisPageRegistrator.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { SectionKind } from 'docc-render/constants/PrimaryContentSection';
+import { anchorize } from 'docc-render/utils/strings';
+import {
+  MainContentSectionAnchors,
+  PrimaryContentSectionAnchors,
+} from 'docc-render/constants/ContentSectionAnchors';
+
+/**
+ * Crawls the `topicData` of a page, and extracts onThisPage sections.
+ */
+export default {
+  watch: {
+    topicData: {
+      immediate: true,
+      handler: 'extractOnThisPageSections',
+    },
+  },
+  methods: {
+    shouldRegisterContentSection(subSection) {
+      return subSection.type === 'heading' && subSection.level < 4;
```

Comment:
Small thing but you may want to export the `BlockType` constant from `ContentNode` so you can use `BlockType.heading` instead of the string literal here.

Reply:
Yep, good catch.